x-pack/plugins/lens/public/datasources/form_based/utils.tsx

```diff
@@ -162,43 +170,63 @@ const accuracyModeEnabledWarning = (columnName: string, docLink: string) => (
   />
 );
 
-export function getTSDBRollupWarningMessages(
+export function getShardFailuresWarningMessages(
```

Comment:
I see that this function is called whether or not the warning is a shard failure. Do we plan to expand the types of failures this function handles? If so, we could consider naming it something more general such as `getSearchWarningMessages`.

Reply:
If I understand the original idea correctly, we should handle different types of warnings separately.

See `x-pack/plugins/lens/public/datasources/form_based/form_based.tsx`

```
getSearchWarningMessages: (state, warning, request, response) => {
  return [...getShardFailuresWarningMessages(state, warning, request, response, core.theme)];
},
```